src/onnx/parse_celu.cpp

```diff
@@ -0,0 +1,53 @@
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/make_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_celu : op_parser<parse_celu>
+{
+    std::vector<op_desc> operators() const { return {{"Celu"}}; }
+
+    instruction_ref parse(const op_desc&,
+                          const onnx_parser&,
+                          const onnx_parser::node_info& info,
+                          std::vector<instruction_ref> args) const
+    {
+        float alpha = 1.0;
+        if(contains(info.attributes, "alpha"))
+        {
+            alpha = info.attributes.at("alpha").f();
+        }
+
+        auto input_lens = args[0]->get_shape().lens();
+        auto input_type = args[0]->get_shape().type();
+        if(input_type != migraphx::shape::float_type)
+        {
+            MIGRAPHX_THROW("CELU: input tensor not float type");
```

Comment:
The onnx spec allows the user to have an input datatype that isn't float. However, alpha is always specified as float. I think we just need to convert the alpha datatype to whatever the input datatype is.

Reply:
It says on the operators.md that there is a type constraint for the tensor T:
Type Constraints
T : tensor(float)
Constrain input and output types to float32 tensors.

Reply:
Oh never mind. I scrolled down further and was reading the wrong operator.